code/modules/telesci/telepads.dm

```diff
@@ -0,0 +1,53 @@
+///SCI TELEPAD///
+/obj/machinery/telepad
+	name = "telepad"
+	desc = "A bluespace telepad used for teleporting objects to and from a location."
+	icon = 'icons/obj/telescience.dmi'
+	icon_state = "pad-idle"
+	anchored = 1
+	use_power = 1
+	idle_power_usage = 200
+	active_power_usage = 5000
+	circuit = /obj/item/weapon/circuitboard/telesci_pad
+	var/efficiency
+
+/obj/machinery/telepad/New()
+	..()
+	component_parts = list(
+	new /obj/item/weapon/circuitboard/telesci_pad,
+	new /obj/item/bluespace_crystal/artificial,
+	new /obj/item/bluespace_crystal/artificial,
+	new /obj/item/weapon/stock_parts/capacitor,
+	new /obj/item/weapon/stock_parts/console_screen
+	)
+	RefreshParts()
+
+/obj/machinery/telepad/RefreshParts()
+	var/E
+	for(var/obj/item/weapon/stock_parts/capacitor/C in component_parts)
+		E += C.rating
+	efficiency = E
+
+/obj/machinery/telepad/attackby(obj/item/I, mob/user, params)
+	if(default_deconstruction(I, user))
+		return
+	if(default_part_replacement(I, user))
+		return
+	if(panel_open)
+		if(istype(I, /obj/item/weapon/tool/multitool))
```

Comment:
any tool interactions should use our tool system, this for example should use QUALITY_PULSING